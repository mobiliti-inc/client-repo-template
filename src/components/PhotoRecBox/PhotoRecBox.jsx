import styles from './PhotoRecBox.scss';

const PhotoRecBox = ({
	onBoxClick,
	boxIcon,
	isVertical,
	boxTitle,
	boxDescription,
	boxImage,
	onUploadImageInputChange
}) => (
	<div role="presentation" onClick={onBoxClick} styleName={`rec-box rec-box--${isVertical ? 'vertical' : 'horizontal'}`} >
		{boxIcon &&
		<span styleName={`circle circle--${isVertical ? 'vertical' : 'horizontal'}`}>
			<img src={boxIcon} alt="box icon" />
		</span>
		}

		{<input
			styleName="input-front-license"
			type="file"
			name="front-license"
			id="front-license"
			accept="image/jpeg"
			onChange={onUploadImageInputChange}

		/>}
		{boxImage &&
		<div styleName="box-image-container">
			<img styleName="box-image" src={boxImage} alt="box" />
			<div styleName="check-mark-circle">
				<span styleName="check-mark" />
			</div>
		</div>}
		{boxTitle && <div styleName={`box-title box-title--${isVertical ? 'vertical' : 'horizontal'}`}>{boxTitle}</div>}
		{boxDescription &&
			<div
				styleName={`box-description box-description--${isVertical ? 'vertical' : 'horizontal'}`}
			>{boxDescription}
			</div>}

	</div >);


PhotoRecBox.propTypes = {
	isVertical: PropTypes.bool,
	boxIcon: PropTypes.string,
	boxTitle: PropTypes.string,
	boxDescription: PropTypes.string,
	onBoxClick: PropTypes.func.isRequired,
	boxImage: PropTypes.string,
	onUploadImageInputChange: PropTypes.func
};

PhotoRecBox.defaultProps = {
	isVertical: true,
	boxIcon: '',
	boxTitle: '',
	boxDescription: '',
	boxImage: '',
	onUploadImageInputChange: () => { }
};

export default CSSModules(PhotoRecBox, styles, { allowMultiple: true });

