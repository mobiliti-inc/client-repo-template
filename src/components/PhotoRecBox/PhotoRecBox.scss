@import "~variables";
.rec-box {
    width: 100%;
    border: 1px solid rgba(0, 0, 0, 0.219);
    border-radius: 5px;
    cursor: pointer;
    display: flex;
    align-items: center;
    justify-content: center;
    &--vertical {
        flex-direction: column;
        padding: 40px 0;
    }
    &--horizontal {
        flex-direction: row;
        padding: 20px 0;
        justify-content: flex-start;
    }
}

.circle {
    height: 66px;
    width: 66px;
    border: 2px solid $darker-blue;
    border-radius: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    &--vertical {
        margin-left: 0;
    }
    &--horizontal {
        margin-left: 10%;
    }
}

.box-title {
    width: 144px;
    color: $darker-blue;
    font-family: $futura;
    font-size: 18px;
    font-weight: bold;
    letter-spacing: -0.4px;
    line-height: 20px;
    text-align: center;
    &--vertical {
        padding: 15px 0 0 0;
    }
    &--horizontal {
        padding: 0 0 0 15px;
    }
}

.box-description {
    width: 173px;
    color: $darker-blue;
    font-family: $hanken-sans;
    font-size: 16px;
    line-height: 19px;
    text-align: center;
    &--vertical {
        padding: 5px 5px;
    }
    &--horizontal {
        padding: 0;
    }
}

.box-image-container {
    margin-left: 10%;
    width: 122px;
    height: 91px;
    display: flex;
    justify-content: center;
    align-items: center;
    position: relative;
    box-shadow: $box-shadow--standard;
}

.box-image {
    width: 100%;
    height: 100%;
    border-radius: 4px;
}

.check-mark-circle {
    position: absolute;
    background-color: $green-check;
    color: white;
    height: 26px;
    width: 26px;
    border-radius: 100%;
    top: -13px;
    right: -13px;
    display: flex;
    justify-content: center;
    align-items: center;
}

.check-mark {
    @include check-mark;
    display: inline-block;
    cursor: pointer;
}

.input-front-license {
    display: none;
}