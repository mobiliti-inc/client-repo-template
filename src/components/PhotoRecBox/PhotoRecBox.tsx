import styles from './PhotoRecBox.scss';

type PhotoRecBoxProps = {
	isVertical?: boolean,
	boxIcon?: string,
	boxTitle?: string,
	boxDescription?: string,
	onBoxClick: (...args: any[]) => any,
	boxImage?: string,
	onUploadImageInputChange?: (...args: any[]) => any
};

const PhotoRecBox: React.SFC<PhotoRecBoxProps> = ({
	onBoxClick,
	boxIcon,
	isVertical,
	boxTitle,
	boxDescription,
	boxImage,
	onUploadImageInputChange
}) => (
		<div role="presentation" onClick={onBoxClick} styleName={`rec-box rec-box--${isVertical ? 'vertical' : 'horizontal'}`}>
			{boxIcon && (
				<span styleName={`circle circle--${isVertical ? 'vertical' : 'horizontal'}`}>
					<img src={boxIcon} alt="box icon" />
				</span>
			)}

			{
				<input
					styleName="input-front-license"
					type="file"
					name="front-license"
					id="front-license"
					accept="image/jpeg"
					onChange={onUploadImageInputChange}
				/>
			}
			{boxImage && (
				<div styleName="box-image-container">
					<img styleName="box-image" src={boxImage} alt="box" />
					<div styleName="check-mark-circle">
						<span styleName="check-mark" />
					</div>
				</div>
			)}
			{boxTitle && <div styleName={`box-title box-title--${isVertical ? 'vertical' : 'horizontal'}`}>{boxTitle}</div>}
			{boxDescription && (
				<div styleName={`box-description box-description--${isVertical ? 'vertical' : 'horizontal'}`}>{boxDescription}</div>
			)}
		</div>
	);

// PhotoRecBox.defaultProps = {
// 	isVertical: true,
// 	boxIcon: '',
// 	boxTitle: '',
// 	boxDescription: '',
// 	boxImage: '',
// 	onUploadImageInputChange: () => {}
// };

export default CSSModules(PhotoRecBox, styles, { allowMultiple: true });
