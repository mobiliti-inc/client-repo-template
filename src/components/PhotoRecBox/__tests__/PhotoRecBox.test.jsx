import PhotoRecBox from '../PhotoRecBox';

describe('PhotoRecBox', () => {
	it('renders with box icon', () => {
		const tree = shallow(
			<PhotoRecBox
				onBoxClick={() => { }}
				boxIcon="test icon"
			/>);
		matchSnapshot(tree);
	});


	it('renders with box image', () => {
		const tree = shallow(
			<PhotoRecBox
				onBoxClick={() => { }}
				boxImage="test image"
			/>);
		matchSnapshot(tree);
	});


	it('renders horizontally when isVertical is set to false', () => {
		const tree = shallow(
			<PhotoRecBox
				onBoxClick={() => { }}
				boxIcon="test icon"
				isVertical={false}
			/>);
		matchSnapshot(tree);
	});

	it('renders with box title if box title is defined', () => {
		const tree = shallow(
			<PhotoRecBox
				onBoxClick={() => { }}
				boxIcon="test icon"
				isVertical={false}
				boxTitle="test title"
			/>);
		matchSnapshot(tree);
	});

	it('renders with box description if box description is defined', () => {
		const tree = shallow(
			<PhotoRecBox
				onBoxClick={() => { }}
				boxIcon="test icon"
				isVertical={false}
				boxDescription="test description"
			/>);
		matchSnapshot(tree);
	});
});
