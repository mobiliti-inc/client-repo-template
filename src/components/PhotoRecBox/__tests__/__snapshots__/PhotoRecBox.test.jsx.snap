// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoRecBox renders horizontally when isVertical is set to false 1`] = `
<div
  className="rec-box rec-box--horizontal"
  onClick={[Function]}
  role="presentation"
>
  <span
    className="circle circle--horizontal"
  >
    <img
      alt="box icon"
      src="test icon"
    />
  </span>
  <input
    accept="image/jpeg"
    className="input-front-license"
    id="front-license"
    name="front-license"
    onChange={[Function]}
    type="file"
  />
</div>
`;

exports[`PhotoRecBox renders with box description if box description is defined 1`] = `
<div
  className="rec-box rec-box--horizontal"
  onClick={[Function]}
  role="presentation"
>
  <span
    className="circle circle--horizontal"
  >
    <img
      alt="box icon"
      src="test icon"
    />
  </span>
  <input
    accept="image/jpeg"
    className="input-front-license"
    id="front-license"
    name="front-license"
    onChange={[Function]}
    type="file"
  />
  <div
    className="box-description box-description--horizontal"
  >
    test description
  </div>
</div>
`;

exports[`PhotoRecBox renders with box icon 1`] = `
<div
  className="rec-box rec-box--vertical"
  onClick={[Function]}
  role="presentation"
>
  <span
    className="circle circle--vertical"
  >
    <img
      alt="box icon"
      src="test icon"
    />
  </span>
  <input
    accept="image/jpeg"
    className="input-front-license"
    id="front-license"
    name="front-license"
    onChange={[Function]}
    type="file"
  />
</div>
`;

exports[`PhotoRecBox renders with box image 1`] = `
<div
  className="rec-box rec-box--vertical"
  onClick={[Function]}
  role="presentation"
>
  <input
    accept="image/jpeg"
    className="input-front-license"
    id="front-license"
    name="front-license"
    onChange={[Function]}
    type="file"
  />
  <div
    className="box-image-container"
  >
    <img
      alt="box"
      className="box-image"
      src="test image"
    />
    <div
      className="check-mark-circle"
    >
      <span
        className="check-mark"
      />
    </div>
  </div>
</div>
`;

exports[`PhotoRecBox renders with box title if box title is defined 1`] = `
<div
  className="rec-box rec-box--horizontal"
  onClick={[Function]}
  role="presentation"
>
  <span
    className="circle circle--horizontal"
  >
    <img
      alt="box icon"
      src="test icon"
    />
  </span>
  <input
    accept="image/jpeg"
    className="input-front-license"
    id="front-license"
    name="front-license"
    onChange={[Function]}
    type="file"
  />
  <div
    className="box-title box-title--horizontal"
  >
    test title
  </div>
</div>
`;
